learn/quickstart/derived_types.md

```diff
@@ -0,0 +1,298 @@
+        [type-bound-procedures]
+end type
+```
+
+### Part 1: Options to declare a derived type
+
+`attribute-list` may refer to the following:
+
+- _access-type_ that is either `public` or `private`
+- `bind(c)` offers interoperability with C programming language 
+- `extends(`_parent_`)` where _parent_ is the name of a previously declared derived type, from which, the current derived type will inherit all its members and functionality.
+- `abstract` an object orianted feature that is covered in the advanced programming tutorial.
+
+{% include note.html content="If the `attribute: bind(c)` or the `statement: sequence` is used then a derived type cannot have the `attribute: extends` and visa-versa." %}
+
+The `sequence` attribute may be used only to declare that the following  members should be accessed in the same order as they are defined within the derived type. 
+
+Example with `sequence`:
+```fortran
+type :: foo
+sequence
+integer :: var1
+real    :: var2
+end type
+! init
+type(foo) :: bar
+bar = foo(1, 0.5)
+```
+{% include note.html content="The use of statement `sequence` presupposes that the data types defined below are neither of `allocatable` nor of `pointer` type. Furthermore, it does not imply that these data types  will be stored in memory in any particular form, there is no relation to `contigeous` attribute." %}
+
+The _access-type_ attributes `public` and `private` if used, declare that all [member-variables] declared below will be automatically assigned the attribute accordingly. 
+
+The attribute `bind(c)` is used to achieve compatibility between Fortran's derived type and C's struct.
+
+Example with `bind(c)`:
+```fortran
+module mymod
+use iso_c_bindings
+implicit none
+type, bind(c) :: mytype
+    integer(c_int) :: i
+end type
+```
+matches the following C struct:
+```c
+struct{
+    int i
+}mytype;
+```
+{% include note.html content="A fortran derived type with the attribute `bind(c)` cannot have the `sequence` and `extends` attributes. Furthermore it cannot contain any Fortran `pointer` or `allocatable` types." %}
+
+`parameterized-decleration-list`: is an optional feautre. If used, then the parameters must be listed in place of [parameterized-definition-statements] and must be either `len` or `kind` parameters or both. 
+
+Example of a derived type with `parameterized-decleration-list` and with the `attribute: public`:
+ ```fortran
+module mymod
+implicit none
+private
+
+type, public :: matrix(rows, cols, k)
+  integer, len :: rows, cols
+  integer, kind :: k = kind(0.0)
+  real(kind = k), dimension(rows, cols) :: values
+end type matrix
+end module
+
+program test
+use mymod
+implicit none
+type(matrix(rows=5, cols=5)) :: m
+end program
+ ```
+{% include note.html content="In this example the parameter **k** has already been assigned a default value of `kind(0.0)`, that is of floating point single precision, for that reason it can be ommited, as it is the case here in the declaration inside the main program." %}
```

Comment:
```suggestion
{% include note.html content="In this example the parameter **k** has already been assigned a default value of `kind(0.0)`, that is of floating point single precision. Therefore, it can be omitted, as it is the case here in the declaration inside the main program." %}
```